ldpd/ldp_zebra.c

```diff
@@ -576,3 +578,17 @@ ldp_zebra_destroy(void)
 	zclient_free(zclient);
 	zclient = NULL;
 }
+
+void ldp_zebra_vrf_register(struct vrf *vrf)
+{
+        if (vrf->vrf_id == VRF_DEFAULT)
+                return;
+        zclient_send_reg_requests(zclient, vrf->vrf_id);
```

Comment:
ldpd only operates on the default VRF, so why should it register to receive information from other VRFs? The same comment applies to several other daemons, like isisd, eigrpd, ospf6d, babeld, nhrpd, etc. This is not only inefficient but can break certain daemons that don't filter messages coming from VRFs other than the default one.

Reply:
Some network gear has VRF support for LDP. This influenced in my decision to support the VRF based registration for LDPD as well as other daemons.

I am open to suggestions in case the community feels that this is not required.

Reply:
Ok. I'll repeat what I said before. ldpd and several other daemons don't have VRF support as of now. Those daemons just assume that all messages they receive from zebra are related the default VRF. So, if you change those daemons to subscribe to receive information from all VRFs of the system, they will misbehave since they will interpret all messages as if they were related to the default VRF. As an example, a Router-ID update message happening on VRF RED would be interpreted as a global Router-ID update by isisd/ospf6d/etc. In order to avoid such problems, you'd need to change the ZAPI callbacks of those daemons to filter-out messages from VRFs other than the default one. But the most logical thing to do is to not request information those daemons don't need, since they only run on the default VRF. That's why I'd like to ask you to update this PR and drop all changes except the ones done on zebra.